### PR TITLE
Add a `[lints]` entry for workspace members missing it

### PR DESCRIPTION
**Pull request by Kanabenki**
# Objective

- Some workspace members do not inherit the global lints.

## Solution

- Add a `[lints]` entry for all files returned by `rg  --files-without-match -F "[lints]" **/Cargo.toml`, except the compile failure tests since these aren't part of the workspace.
- Add some docstrings where needed.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

**Review by BD103 — APPROVED** [review]

**Review by james7132 — APPROVED** [review]

**Review by tguichaoua — APPROVED** [review]

**Review by doonv — APPROVED** [review]

**Comment by james7132** [comment]
`error: failed to parse manifest at `/home/runner/work/bevy/bevy/benches/Cargo.toml`

We may want to remove this if this is causing a failure.

File: benches/Cargo.toml [hunk 1]
```diff
@@ -6,6 +6,9 @@ description = "Benchmarks for Bevy engine"
 publish = false
 license = "MIT OR Apache-2.0"
 
+[lints]
```
Comment (alice-i-cecile) [inline benches/Cargo.toml:9]:
```suggestion
```

File: benches/Cargo.toml [hunk 2]
```diff
@@ -6,6 +6,9 @@ description = "Benchmarks for Bevy engine"
...
+workspace = true
```
Comment (alice-i-cecile) [inline benches/Cargo.toml:10]:
```suggestion
```

File: benches/Cargo.toml [hunk 3]
```diff
@@ -6,6 +6,9 @@ description = "Benchmarks for Bevy engine"
...
+
```
Comment (alice-i-cecile) [inline benches/Cargo.toml:11]:
```suggestion
```

**Comment by Kanabenki** [comment]
> `` error: failed to parse manifest at `/home/runner/work/bevy/bevy/benches/Cargo.toml` `

Missed that it wasn't part of the workspace. @alice-i-cecile thanks for taking care of it!

**Pull request merged**